addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -1175,6 +1163,18 @@ function makeActionManager(env) {
         // END LEGACY CODE COMPATIBILITY
 
         newController.props = _getViewProps(view, controller.action, controller.views, props);
+
+        let sharedState;
+        if (controller && controller.getSharedState) {
+            sharedState = controller.getSharedState();
+            controller.sharedState = sharedState;
+        }
+        if (sharedState) {
+            newController.props.sharedState = sharedState;
+        } else if (newController.sharedState) {
+            newController.props.sharedState = newController.sharedState;
+        }
```

Comment:
also, not sure about the logic here. Here is how I see it:
- when leaving a view, we ask it to export the shared state: say `{ domain: [] }`
- this state is stored on the action, and given in props to the new view
- when the new view is left, it exports its shared state as well, say `{ res_ids: [] }` (it omits `domain` because it doesn't care of that)
- we should ensure that we don't lose the previously exported state, so we need to merge the state we previously stored on the action with the newly exported one
- we store again that state on the action and give it in props to the new view

The alternative would've been to force each view to always export all keys it receives, even if it doesn't know what these keys are about. That way, no logic needed in action service, no need to store anything. But it would imply that view must always export the whole shared state, because if a view doesn't do it, information is lost as soon as we travel through that view.

As I said on Friday, I don't have a strong opinion on which alternative we should implement.

Reply:
we go for option 1: do stuff in action service

store things on controller.action + object.assign